components/server/src/ome/services/ThumbnailBean.java

```diff
@@ -1070,6 +1079,35 @@ protected void actOnOneGroup(Set<Long> pixelsIds) {
         return value;
     }
 
+    /*
+     * (non-Javadoc)
+     *
+     * @see ome.api.ThumbnailStore#getThumbnail(ome.model.core.Pixels,
+     *      ome.model.displayRenderingDef, java.lang.Integer,
+     *      java.lang.Integer)
+     */
+    @RolesAllowed("user")
+    @Transactional(readOnly = false)
+    public byte[] getThumbnailWithoutDefault(Integer sizeX, Integer sizeY) {
+        errorIfNullPixelsAndRenderingDef();
+        Dimension dimensions = sanityCheckThumbnailSizes(sizeX, sizeY);
+        Set<Long> pixelsIds = Collections.singleton(pixelsId);
+        ctx.loadAndPrepareMetadata(pixelsIds, dimensions);
+
+        // If this comes back null, don't have a thumbnail yet
+        thumbnailMetadata = ctx.getMetadataSimple(pixelsId);
+        if (thumbnailMetadata == null) {
+            // We don't have a thumbnail to load, lets try to create it
+            // and then return it
+            thumbnailMetadata = ctx.createThumbnailMetadata(pixels, dimensions);
+        }
+
+        byte[] value = retrieveThumbnail(thumbnailMetadata);
+        // I don't really know why this is here, no iquery calls being that I can see...
```

Comment:
This has to do with the Hibernate/transaction handling of this service since it's stateful.